scripts/partition_manager.py

```diff
@@ -565,22 +614,42 @@ def parse_args():
                         help="List of paths to input yaml files. ")
 
-    parser.add_argument("--flash-size", required=True, type=int,
-                        help="Flash size of chip in kB.")
+    parser.add_argument("--output-partitions", required=True, type=str,
+                        help="Path to output partition configuration file.")
+
+    parser.add_argument("--output-regions", required=True, type=str,
+                        help="Path to output regions configuration file.")
 
-    parser.add_argument("--flash-start", type=lambda x: int(x, 0), default=0,
-                        help="Start address of flash.")
+    parser.add_argument("--static-config", required=False, type=argparse.FileType(mode='r'),
+                        help="Path static configuration.")
 
-    parser.add_argument("--output", required=True, type=str,
-                        help="Path to output file.")
+    parser.add_argument("--regions", required=False, type=str, nargs='*',
+                        help="Space separated list of regions. For each region specified here, one must specify"
+                             "--{region_name}-base-addr and --{region_name}-size. If the region is associated"
+                             "with a driver, the device name must be given in --{region_name}-device (e.g. an "
+                             "external flash driver. For regions with placement strategy 'complex' it is possible"
+                             "to specify the --{region_name}-dynamic-partition to set the name of the dynamic partition"
+                             "which occupy all non-used area.")
 
-    parser.add_argument("-d", "--dynamic-partition", required=False, type=str,
-                        help="Name of dynamic partition")
+    known, ranges_arguments = parser.parse_known_args()
 
-    parser.add_argument("-s", "--static-config", required=False, type=argparse.FileType(mode='r'),
-                        help="Path static configuration.")
+    # Create new instance to parse regions
+    parser = argparse.ArgumentParser()
+    for x in known.regions:
+        # Generate arguments for each region dynamically
+        parser.add_argument(f'--{x}-size', required=True, type=lambda z: int(z, 0))
+        parser.add_argument(f'--{x}-base-address', required=False, type=lambda z: int(z, 0), default=0)
+        parser.add_argument(f'--{x}-placement-strategy', required=False, type=str,
+                            choices=[START_TO_END, END_TO_START, COMPLEX], default=START_TO_END)
+        parser.add_argument(f'--{x}-device', required=False, type=str, default='')
+        parser.add_argument(f'-{x}-dynamic-partition', required=False, type=str, help="Name of dynamic partition")
+
+    ranges_configuration = parser.parse_args(ranges_arguments)
+
+    # Merge the known parameters with the ranges parameters
+    known.__dict__.update({k: v for k, v in vars(ranges_configuration).items()})
```

Comment:
Would it make sense instead to `return (known, ranges_configuration)`? I ask since they seem to be separated later on anyways.

Reply:
I agree